src/test/groovy/org/osiam/test/integration/RegistrationIT.groovy

```diff
@@ -152,6 +165,17 @@ class RegistrationIT extends AbstractIT {
 
         then:
         responseStatus == 200
+
+        //Waiting at least 5 seconds to ensure that an email was send
+        Thread.sleep(5000)
```

Comment:
you could actively poll for new emails. No need for a latch, if you don't poll in a background thread.